hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -284,14 +284,29 @@
             = new HazelcastProperty("hazelcast.max.join.seconds", 300, SECONDS);
     public static final HazelcastProperty MAX_JOIN_MERGE_TARGET_SECONDS
             = new HazelcastProperty("hazelcast.max.join.merge.target.seconds", 20, SECONDS);
+
+    /** The interval at which heartbeats are sent and received heartbeats and master confirmations are checked */
     public static final HazelcastProperty HEARTBEAT_INTERVAL_SECONDS
             = new HazelcastProperty("hazelcast.heartbeat.interval.seconds", 5, SECONDS);
+    /**
+     * The timeout which defines when the received heartbeat is ignored, when the heartbeat timestamps are reset because
+     * of cluster clock jumps and when a cluster member is removed because it has not sent any heartbeats.
+     */
```

Comment:
Mainly this property defines maximum timeout of heartbeat  to declare a member as dead. Other effects (ignoring heartbeats, clock jumps etc) are impl details.